vipermonkey/core/expressions.py

```diff
@@ -2088,12 +2109,19 @@ def _handle_regex_execute(self, context, tmp_lhs):
             return None
 
         # Find all the regex matches in the string.
-        r = None
         try:
-            r = re.findall(pat, mod_str)
+            matches = re.finditer(pat, mod_str)
+            match_objects = []
+            for match in matches:
```

Comment:
No need to a temporary variable for `matches`, `for match in re.finditer(pat, mod_str):` does the same thing directly. And if someone tries to reuse `matches` it would be incorrect because the iterator would be exhausted.